kernel: bump 6.12 to 6.12.43

graysky2 opened the pull request:
Kernel updates through 6.12.43

graysky2 committed:
kernel: bump 6.12 to 6.12.41

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.12.41

All patches automatically rebased.

Build system: x86/64
Build-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64-glibc
Run-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64-glibc

Signed-off-by: John Audia <therealgraysky@proton.me>
Link: https://github.com/openwrt/openwrt/pull/19618
Signed-off-by: Robert Marko <robimarko@gmail.com>

mcprat commented:
lets hold on for just one more day until we figure out what to do in #19490

graysky2 commented:
@hauke - suggested fix?

hauke commented:
These two patches rename `qpic_v2` to `qpic_version2`:
target/linux/generic/backport-6.6/413-01-v6.14-mtd-rawnand-qcom-cleanup-qcom_nandc-driver.patch
target/linux/generic/backport-6.12/410-01-v6.14-mtd-rawnand-qcom-cleanup-qcom_nandc-driver.patch
They should also also adapt these new change:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/diff/drivers/mtd/nand/raw/qcom_nandc.c?id=v6.12.41&id2=v6.12.40

graysky2 commented:
@hauke - I can make the rename change but adapting the changes in the diff you showed is a bit above me.

qosmio commented:
> @hauke - I can make the rename change but adapting the changes in the diff you showed is a bit above me.

The change @hauke mentioned in that diff is actually what is causing the issue. `qpic_v2` vs `qpic_version2`.

FYI, here's the diff of your PR after applying the fix to work on `qualcommax`

```diff
diff --git a/target/linux/generic/backport-6.12/410-01-v6.14-mtd-rawnand-qcom-cleanup-qcom_nandc-driver.patch b/target/linux/generic/backport-6.12/410-01-v6.14-mtd-rawnand-qcom-cleanup-qcom_nandc-driver.patch
index 95b66071ad..1d1288e832 100644
--- a/target/linux/generic/backport-6.12/410-01-v6.14-mtd-rawnand-qcom-cleanup-qcom_nandc-driver.patch
+++ b/target/linux/generic/backport-6.12/410-01-v6.14-mtd-rawnand-qcom-cleanup-qcom_nandc-driver.patch
@@ -815,7 +815,15 @@ Signed-off-by: Miquel Raynal <miquel.raynal@bootlin.com>
  	read_reg_dma(nandc, NAND_FLASH_STATUS, 1, NAND_BAM_NEXT_SGL);
  
  	ret = submit_descs(nandc);
-@@ -2864,7 +2815,7 @@ static int qcom_param_page_type_exec(str
+@@ -2862,14 +2813,14 @@ static int qcom_param_page_type_exec(str
+ 
+ 	reg_base = NAND_READ_LOCATION_0;
+ 
+-	if (nandc->props->qpic_v2)
++	if (nandc->props->qpic_version2)
+ 		reg_base = NAND_READ_LOCATION_LAST_CW_0;
+ 
+ 	ret = qcom_parse_instructions(chip, subop, &q_op);
  	if (ret)
  		return ret;
  
@@ -824,7 +832,7 @@ Signed-off-by: Miquel Raynal <miquel.raynal@bootlin.com>
  
  	nandc->buf_count = 0;
  	nandc->buf_start = 0;
-@@ -2872,38 +2823,38 @@ static int qcom_param_page_type_exec(str
+@@ -2877,52 +2828,52 @@ static int qcom_param_page_type_exec(str
  	clear_read_regs(nandc);
  	clear_bam_transaction(nandc);
  
@@ -890,9 +898,14 @@ Signed-off-by: Miquel Raynal <miquel.raynal@bootlin.com>
  	}
  
  	instr = q_op.data_instr;
-@@ -2912,9 +2863,9 @@ static int qcom_param_page_type_exec(str
+ 	op_id = q_op.data_instr_idx;
+ 	len = nand_subop_get_data_len(subop, op_id);
  
- 	nandc_set_read_loc(chip, 0, 0, 0, len, 1);
+-	if (nandc->props->qpic_v2)
++	if (nandc->props->qpic_version2)
+ 		nandc_set_read_loc_last(chip, reg_base, 0, len, 1);
+ 	else
+ 		nandc_set_read_loc_first(chip, reg_base, 0, len, 1);
  
 -	if (!nandc->props->qpic_v2) {
 -		write_reg_dma(nandc, NAND_DEV_CMD_VLD, 1, 0);
@@ -903,7 +916,7 @@ Signed-off-by: Miquel Raynal <miquel.raynal@bootlin.com>
  	}
  
  	nandc->buf_count = 512;
-@@ -2926,9 +2877,10 @@ static int qcom_param_page_type_exec(str
+@@ -2934,9 +2885,10 @@ static int qcom_param_page_type_exec(str
  		      nandc->buf_count, 0);
  
  	/* restore CMD1 and VLD regs */
@@ -917,7 +930,7 @@ Signed-off-by: Miquel Raynal <miquel.raynal@bootlin.com>
  	}
  
  	ret = submit_descs(nandc);
-@@ -3017,7 +2969,7 @@ static const struct nand_controller_ops
+@@ -3025,7 +2977,7 @@ static const struct nand_controller_ops
  
  static void qcom_nandc_unalloc(struct qcom_nand_controller *nandc)
  {
@@ -926,7 +939,7 @@ Signed-off-by: Miquel Raynal <miquel.raynal@bootlin.com>
  		if (!dma_mapping_error(nandc->dev, nandc->reg_read_dma))
  			dma_unmap_single(nandc->dev, nandc->reg_read_dma,
  					 MAX_REG_RD *
-@@ -3070,7 +3022,7 @@ static int qcom_nandc_alloc(struct qcom_
+@@ -3078,7 +3030,7 @@ static int qcom_nandc_alloc(struct qcom_
  	if (!nandc->reg_read_buf)
  		return -ENOMEM;
  
@@ -935,7 +948,7 @@ Signed-off-by: Miquel Raynal <miquel.raynal@bootlin.com>
  		nandc->reg_read_dma =
  			dma_map_single(nandc->dev, nandc->reg_read_buf,
  				       MAX_REG_RD *
-@@ -3151,15 +3103,15 @@ static int qcom_nandc_setup(struct qcom_
+@@ -3159,15 +3111,15 @@ static int qcom_nandc_setup(struct qcom_
  	u32 nand_ctrl;
  
  	/* kill onenand */
@@ -954,7 +967,7 @@ Signed-off-by: Miquel Raynal <miquel.raynal@bootlin.com>
  		nand_ctrl = nandc_read(nandc, NAND_CTRL);
  
  		/*
-@@ -3176,7 +3128,7 @@ static int qcom_nandc_setup(struct qcom_
+@@ -3184,7 +3136,7 @@ static int qcom_nandc_setup(struct qcom_
  	}
  
  	/* save the original values of these registers */
@@ -963,7 +976,7 @@ Signed-off-by: Miquel Raynal <miquel.raynal@bootlin.com>
  		nandc->cmd1 = nandc_read(nandc, dev_cmd_reg_addr(nandc, NAND_DEV_CMD1));
  		nandc->vld = NAND_DEV_CMD_VLD_VAL;
  	}
-@@ -3349,7 +3301,7 @@ static int qcom_nandc_parse_dt(struct pl
+@@ -3357,7 +3309,7 @@ static int qcom_nandc_parse_dt(struct pl
  	struct device_node *np = nandc->dev->of_node;
  	int ret;
  
@@ -972,7 +985,7 @@ Signed-off-by: Miquel Raynal <miquel.raynal@bootlin.com>
  		ret = of_property_read_u32(np, "qcom,cmd-crci",
  					   &nandc->cmd_crci);
  		if (ret) {
-@@ -3474,30 +3426,30 @@ static void qcom_nandc_remove(struct pla
+@@ -3482,30 +3434,30 @@ static void qcom_nandc_remove(struct pla
  
  static const struct qcom_nandc_props ipq806x_nandc_props = {
  	.ecc_modes = (ECC_RS_4BIT | ECC_BCH_8BIT),

```

zxlhhyccc commented:
It seems that there is a problem with the gpg certificate.

mcprat commented:
@hauke can we please make a decision on #19490 before the next bump is approved?

qosmio commented:
> @graysky2 
> need to be manually changed back to `qpic_v2`?

Sorry, for the poor wording, I could've worded it better.

[With 6.14.41 another block of code was added that referenced qpic_v2](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/diff/drivers/mtd/nand/raw/qcom_nandc.c?id=v6.12.41&id2=v6.12.40)
```diff
--- a/drivers/mtd/nand/raw/qcom_nandc.c
+++ b/drivers/mtd/nand/raw/qcom_nandc.c
@@ -2858,7 +2858,12 @@ static int qcom_param_page_type_exec(struct nand_chip *chip,  const struct nand_
 	const struct nand_op_instr *instr = NULL;
 	unsigned int op_id = 0;
 	unsigned int len = 0;
-	int ret;
+	int ret, reg_base;
+
+	reg_base = NAND_READ_LOCATION_0;
+
+	if (nandc->props->qpic_v2)
+		reg_base = NAND_READ_LOCATION_LAST_CW_0;
 
 	ret = qcom_parse_instructions(chip, subop, &q_op);
 	if (ret)
@@ -2910,7 +2915,10 @@ static int qcom_param_page_type_exec(struct nand_chip *chip,  const struct nand_
 	op_id = q_op.data_instr_idx;
 	len = nand_subop_get_data_len(subop, op_id);
 
-	nandc_set_read_loc(chip, 0, 0, 0, len, 1);
+	if (nandc->props->qpic_v2)
+		nandc_set_read_loc_last(chip, reg_base, 0, len, 1);
+	else
+		nandc_set_read_loc_first(chip, reg_base, 0, len, 1);
 
 	if (!nandc->props->qpic_v2) {
 		write_reg_dma(nandc, NAND_DEV_CMD_VLD, 1, 0);
```

The patch just needed to account for the extra `qpic_v2` references.

I did the following:
```sh
owrt@builder [linux-6.12.41]
❯ quilt push generic-backport/410-01-v6.14-mtd-rawnand-qcom-cleanup-qcom_nandc-driver.patch
❯ sed -i 's/qpic_v2/qpic_version2/g' drivers/mtd/nand/raw/qcom_nandc.c
❯ quilt refresh --strip-trailing-whitespace
❯ quilt push --refresh -a
```

qosmio commented:
@graysky2 sure, [410-01-v6.14-mtd-rawnand-qcom-cleanup-qcom_nandc-driver.patch](https://gist.github.com/qosmio/b11823606a0b873a485d34dbef2d8b39#file-410-01-v6-14-mtd-rawnand-qcom-cleanup-qcom_nandc-driver-patch)

graysky2 committed:
kernel: bump 6.12 to 6.12.42

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.12.42

Added backport to fix ipv6 breakage with the 6.12.42 release:
generic/backport-6.12/621-proc-fix-missing-pde_set_flags.patch[1]

All other patches automatically rebased.

1. https://lore.kernel.org/all/20250821105806.1453833-1-wangzijie1@honor.com

Build system: x86/64
Build-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64-glibc
Run-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64-glibc (Intel N150)

Signed-off-by: John Audia <therealgraysky@proton.me>
Link: https://github.com/openwrt/openwrt/pull/19618
Signed-off-by: Robert Marko <robimarko@gmail.com>

graysky2 committed:
generic/config-6.12: add new option

Add new option required to build with VHOST_MENU=y

This option enables two IOCTLs: VHOST_SET_FORK_FROM_OWNER and
VHOST_GET_FORK_FROM_OWNER. These allow userspace applications
to modify the vhost worker mode for vhost devices.

Also expose module parameter 'fork_from_owner_default' to allow users
to configure the default mode for vhost workers.

By default, `VHOST_ENABLE_FORK_OWNER_CONTROL` is set to `y`,
users can change the worker thread mode as needed.
If this config is disabled (n),the related IOCTLs and parameters will
be unavailable.

If unsure, say "Y".

Signed-off-by: John Audia <therealgraysky@proton.me>
Link: https://github.com/openwrt/openwrt/pull/19618
Signed-off-by: Robert Marko <robimarko@gmail.com>

rmandrad commented:
tested on mediatek/filogic banana pi4

mgz0227 commented:
I had no problem with ipv6 in version 6.12.41. After updating to version 6.12.42, I got the public ipv6 but the public network is inaccessible. I don't know why. I am using an x86/64 machine.

rockdrilla commented:
I'd recommend to check twice recent commits against `netifd`.

mgz0227 commented:
The following commits may cause https://github.com/openwrt/openwrt/pull/19618#issuecomment-3194091252 error:

[ipv6: fix possible infinite loop in fib6_info_uses_dev()](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.42&id=db65739d406c72776fbdbbc334be827ef05880d2)
[ipv6: prevent infinite loop in rt6_nlmsg_size()](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.42&id=cd8d8bbd9ced4cc5d06d858f67d4aa87745e8f38)
[ipv6: annotate data-races around rt->fib6_nsiblings](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.42&id=0c58f74f8aa991c2a63bb58ff743e1ff3d584b62
)

In Linux kernel 6.12.42:
IPv6 FIB (Forwarding Information Base) multipath and concurrent access handling was made stricter (READ_ONCE / WRITE_ONCE + RCU retry).

The RA (Router Advertisement) “automatic” mode relies on checking whether a local default route exists.

With the stricter FIB handling, this check can fail in multipath scenarios → the automatic mode does not advertise a default route → client IPv6 connectivity fails.

Solution:
On the OpenWrt br-lan interface, change the IPv6 RA default router mode from:
Automatic → Use available prefixes
RA does not depend on the local default route check
As long as a public IPv6 prefix exists, the default route is advertised
other clients receive the correct default route

gerardr1 commented:
> @graysky2 friendly ping, 6.12.43 is just released.(｡･ω･｡)

Hi! I have 6.12.43 running on a qualcommax/ipq807x Dynalink router. I had to apply the qpic_v2 rename patches mentioned above, and delete the patches below, if it's of any help:

`target/linux/generic/backport-6.12/410-01-v6.14-mtd-rawnand-qcom-cleanup-qcom_nandc-driver.patch
target/linux/generic/backport-6.12/612-07-v6.17-net-dsa-b53-prevent-SWITCH_CTRL-access-on-BCM5325.patch
target/linux/generic/backport-6.12/612-08-v6.17-net-dsa-b53-fix-IP_MULTICAST_CTRL-on-BCM5325.patch
target/linux/generic/backport-6.12/612-09-v6.17-net-dsa-b53-prevent-DIS_LEARNING-access-on-BCM5325.patch
target/linux/generic/backport-6.12/612-11-v6.17-net-dsa-b53-prevent-GMII_PORT_OVERRIDE_CTRL-access-on-BCM5325.patch
target/linux/generic/backport-6.12/612-14-v6.17-net-dsa-b53-ensure-BCM5325-PHYs-are-enabled.patch
target/linux/generic/backport-6.12/830-v6.17-Revert-leds-trigger-netdev-Configure-LED-blink-inter.patch
target/linux/qualcommax/patches-6.12/0170-clk-qcom-gcc-ipq8074-fix-broken-freq-table-for-nss_port6_tx_clk_src.patch`

rsalvaterra commented:
> The following commits may cause [#19618 (comment)](https://github.com/openwrt/openwrt/pull/19618#issuecomment-3194091252) error:
> 
> [ipv6: fix possible infinite loop in fib6_info_uses_dev()](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.42&id=db65739d406c72776fbdbbc334be827ef05880d2) [ipv6: prevent infinite loop in rt6_nlmsg_size()](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.42&id=cd8d8bbd9ced4cc5d06d858f67d4aa87745e8f38) [ipv6: annotate data-races around rt->fib6_nsiblings](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.42&id=0c58f74f8aa991c2a63bb58ff743e1ff3d584b62)

Not at all. I bisected it down to this one, https://github.com/gregkh/linux/commit/fc1072d934f687e1221d685cf1a49a5068318f34, and I'm still a bit shocked. Already reported it upstream.

mgz0227 commented:
> I'm pretty sure it will be. I already tested with 6.12.43 (albeit pristine, without any OpenWrt patches).

That's great, I'll compile a copy on my openwrt (｡･ω･｡)

graysky2 committed:
kernel: bump 6.12 to 6.12.43

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.12.43

Removed upstreamed:
    generic/backport-6.12/612-07-v6.17-net-dsa-b53-prevent-SWITCH_CTRL-access-on-BCM5325.patch[1]
    generic-backport/612-08-v6.17-net-dsa-b53-fix-IP_MULTICAST_CTRL-on-BCM5325.patch[2]
    generic-backport/612-09-v6.17-net-dsa-b53-prevent-DIS_LEARNING-access-on-BCM5325.patch[3]
    generic-backport/612-11-v6.17-net-dsa-b53-prevent-GMII_PORT_OVERRIDE_CTRL-access-on-BCM5325.patch[4]
    generic-backport/612-14-v6.17-net-dsa-b53-ensure-BCM5325-PHYs-are-enabled.patch[5]
    generic-backport/830-v6.17-Revert-leds-trigger-netdev-Configure-LED-blink-inter.patch[6]
    bcm27xx/patches-6.12/950-0100-media-tc358743-Increase-FIFO-level-to-374.patch[7]
    bcm27xx/patches-6.12/950-0102-media-tc358743-Check-I2C-succeeded-during-probe.patch[8]
    bcm27xx/patches-6.12/950-0107-media-tc358743-Return-an-appropriate-colorspace-from.patch[9]
    bcm27xx/patches-6.12/950-0341-Allow-RESET_BRCMSTB-on-ARCH_BCM2835.patch[10]
    lantiq/patches-6.12/102-11-v6.17-MIPS-lantiq-falcon-sysctrl-fix-request-memory-check-.patch[11]
    lantiq/patches-6.12/102-12-v6.17-MIPS-vpe-mt-add-missing-prototypes-for-vpe_-alloc-st.patch[12]
    qualcommax/patches-6.12/0170-clk-qcom-gcc-ipq8074-fix-broken-freq-table-for-nss_port6_tx_clk_src.patch[13]
    generic/backport-6.12/612-13-v6.17-net-dsa-b53-fix-b53_imp_vlan_setup-for-BCM5325.patch[14]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.43&id=f16f96ccfe56ef2d627c20ba47f6d9d7dea40f5b
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.43&id=394bd12d1f93ed2fa0fa1d8f31ffd3c344681e85
3. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.43&id=9f45e4858f8096bb27f9a2c75fc24a538dffad67
4. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.43&id=15cf46cc6ecc7a54b397294675d11c379ddf69ef
5. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.43&id=0d250ad617a035904744cef74a5d5dc43d0199b9
6. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.43&id=cb9bb872366e10fa42e1e0200a0c2e5d05a27bec
7. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.43&id=f25d566c5674decd0051a3e624805fc1524729bf
8. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.43&id=3e03ee3b65baeac4c253b30650492ffffad9b608
9. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.43&id=b42b107293b52ac8d46238815b1071138a47a7e4
10. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.43&id=e285cdb9d2e4ba2e4515ae898f92e9f38be64eb3
11. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.43&id=1e9079ff83eac962bc3b1e2fbad73b2f7d5256f4
12. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.43&id=8520c843d9a2770012f23da08e4e2ccf1bb05ba6
13. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.43&id=f1c5c55fb6cda312f5ed15505588caaf05a69043
14. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.43&id=757955c80663ac3df7053abaf6fcc75051ef7a7c

Build system: x86/64 (Intel N150 based)
Build-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64-glibc
Run-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64-glibc

Signed-off-by: John Audia <therealgraysky@proton.me>
Link: https://github.com/openwrt/openwrt/pull/19618
Signed-off-by: Robert Marko <robimarko@gmail.com>

graysky2 commented:
It's there now

mgz0227 commented:
> It's there now

Good, I can see it now. Thanks for updating.

DragonBluep commented on target/linux/generic/backport-6.12/612-13-v6.17-net-dsa-b53-fix-b53_imp_vlan_setup-for-BCM5325.patch:
```diff
@@ -19,9 +19,9 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
+ 		cpu_port = B53_CPU_PORT;
  
 +	/* BCM5325 CPU port is at 8 */
 +	if ((is5325(dev) || is5365(dev)) && cpu_port == B53_CPU_PORT_25)
```
This patch can be removed.

graysky2 replied:
done, marked as number 14

hauke commented on target/linux/generic/backport-6.12/612-13-v6.17-net-dsa-b53-fix-b53_imp_vlan_setup-for-BCM5325.patch:
Please link this commit in the commit message:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.43&id=757955c80663ac3df7053abaf6fcc75051ef7a7c

graysky2 commented on target/linux/generic/backport-6.12/621-proc-fix-missing-pde_set_flags.patch:
fixed

rsalvaterra replied:
Either I'm missing something, or `621-proc-fix-missing-pde_set_flags.patch` is still part of the 6.12.43 bump, not 6.12.42, as it should be.

hauke replied:
It is still part of the 6.6.43 commit

graysky2 replied:
Man, third time's a charm

EPinci commented:
Urgh. Main with this merged seems to have broken my BPI R4 boot from eMMC...

graysky2 commented:
> Lets see what breaks

That didn't take long! 36 minutes :smile:

EPinci commented:
Ok, turns out the device is booting but my SFP 10gb uplink NIC is dead.
I think it ties back to https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=57a127c9e79db96f4dbf262db9e1a808bd474ddf
Manually adding back the aquantia drivers brought me back online.

graysky2 commented:
Good to hear it wasn't something I introduced :smile:
